app/src/ui/lib/text-box.tsx

```diff
@@ -177,6 +177,14 @@ export class TextBox extends React.Component<ITextBoxProps, ITextBoxState> {
     )
   }
 
+  private onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
Now that we've moved TextBox to not expose the onChange event directly I think we should move the Escape check in here instead.

```ts
if (event.key === 'Escape' && this.props.type === 'search') {
  if (this.inputElement && this.inputElement.value.length > 0) {
    event.preventDefault()
    this.props.onValueChanged('')
  }
}
```

Reply:
`onEscape`?

Reply:
I initially was going to put it in that component, then I thought about the reason why we use `Esc` to clear the contents of the textbox and decided it should be in the filter list. It seems that we only want to `Esc`to clear the contents of in filter-lists and no where else. Is that an incorrect assumption on my part?

Reply:
@iAmWillShepherd The way I look at it is it's a behavior of a search input field. Like, if we were to use the `TextBox` with `type=search` somewhere outside of a filter list we'd still want to clear its text when you hit escape. It just so happens to be the case that we only use search textboxes inside of filter lists at the moment.

> onEscape

@j-f1 We try to match event handler names to event names so it should be `onKeyDown`

Reply:
@niik I was suggesting having an event that only gets called when the escape key is pressed.